oap-server/server-cluster-plugin/cluster-consul-plugin/src/test/java/org/apache/skywalking/oap/server/cluster/plugin/consul/ITClusterModuleConsulProviderFunctionalTest.java

```diff
@@ -0,0 +1,266 @@
+
+        List<RemoteInstance> remoteInstancesOfA = queryRemoteNodes(providerA, 2);
+        validateServiceInstance(addressA, addressB, remoteInstancesOfA);
+
+        List<RemoteInstance> remoteInstancesOfB = queryRemoteNodes(providerB, 2);
+        validateServiceInstance(addressB, addressA, remoteInstancesOfB);
+    }
+
+    @Test
+    public void unregisterRemoteOfCluster() throws Exception {
+        final String serviceName = "unregister_remote_cluster";
+        ModuleProvider providerA = createProvider(serviceName);
+        ModuleProvider providerB = createProvider(serviceName);
+
+        Address addressA = new Address("127.0.0.6", 1006, true);
+        Address addressB = new Address("127.0.0.7", 1007, true);
+
+        RemoteInstance instanceA = new RemoteInstance(addressA);
+        RemoteInstance instanceB = new RemoteInstance(addressB);
+
+        getClusterRegister(providerA).registerRemote(instanceA);
+        getClusterRegister(providerB).registerRemote(instanceB);
+
+        List<RemoteInstance> remoteInstancesOfA = queryRemoteNodes(providerA, 2);
+        validateServiceInstance(addressA, addressB, remoteInstancesOfA);
+
+        List<RemoteInstance> remoteInstancesOfB = queryRemoteNodes(providerB, 2);
+        validateServiceInstance(addressB, addressA, remoteInstancesOfB);
+
+        // unregister A
+        Consul client = Whitebox.getInternalState(providerA, "client");
+        AgentClient agentClient = client.agentClient();
+        agentClient.deregister(instanceA.getAddress().toString());
+
+        // only B
+        remoteInstancesOfB = queryRemoteNodes(providerB, 1, 120);
+        assertEquals(1, remoteInstancesOfB.size());
+        Address address = remoteInstancesOfB.get(0).getAddress();
+        assertEquals(address, addressB);
+        assertTrue(addressB.isSelf());
+    }
+
+    private ClusterModuleConsulProvider createProvider(String serviceName) throws Exception {
+        return createProvider(serviceName, null, 0);
+    }
+
+    private ClusterModuleConsulProvider createProvider(String serviceName, String internalComHost, int internalComPort) throws Exception {
+        ClusterModuleConsulProvider provider = new ClusterModuleConsulProvider();
+
+        ClusterModuleConsulConfig config = (ClusterModuleConsulConfig) provider.createConfigBeanIfAbsent();
+
+        config.setHostPort(consulAddress);
+        config.setServiceName(serviceName);
+
+        if (!StringUtil.isEmpty(internalComHost)) {
+            config.setInternalComHost(internalComHost);
+        }
+
+        if (internalComPort > 0) {
+            config.setInternalComPort(internalComPort);
+        }
+
+        provider.prepare();
+        provider.start();
+        provider.notifyAfterCompleted();
+
+        ConsulCoordinator consulCoordinator = (ConsulCoordinator) provider.getService(ClusterRegister.class);
+        // ignore health check
+        ClusterRegister register = new ClusterRegister() {
```

Comment:
Replace with lambda

Reply:
```remoteInstance``` needs to be reassigned

Reply:
> `remoteInstance` needs to be reassigned

Doesn't it compile? It works well on my machine
![image](https://user-images.githubusercontent.com/15965696/61297542-06964d00-a80f-11e9-9cbe-a71e33ba1d1c.png)

Reply:
done